src/screens/Settings/components/Settings.tsx

```diff
@@ -0,0 +1,154 @@
+
+const styles = EStyleSheet.create({
+	container: {
+		backgroundColor: '$gray100',
+		flex: 1,
+		alignItems: 'center',
+	},
+	switch: {
+		alignItems: 'center',
+		justifyContent: 'space-between',
+		flexDirection: 'row',
+		borderBottomWidth: 1,
+		borderColor: '$gray300',
+		marginHorizontal: 20,
+		padding: 20,
+		width: 250,
+	},
+	'switch:last-child': {
+		borderBottomWidth: 0,
+	},
+	userIcon: {
+		width: 100,
+		height: 100,
+		borderRadius: 50,
+	},
+	user: {
+		flexDirection: 'row',
+		alignItems: 'center',
+		justifyContent: 'center',
+		borderBottomWidth: 1,
+		borderBottomColor: '$gray300',
+		paddingVertical: 20,
+		width: '100%',
+	},
+	settingsBox: {
+		width: Dimensions.get('screen').width - 40,
+		backgroundColor: '$white',
+		borderColor: '$gray300',
+		alignItems: 'center',
+		justifyContent: 'center',
+		borderRadius: 10,
+		borderWidth: 1,
+		height: 170,
+		marginVertical: 20,
+	},
+	settingsBoxHeader: {
+		display: 'flex',
+		justifyContent: 'center',
+		color: '$gray600',
+		marginTop: 10,
+		fontSize: 16,
+	},
+	passwordInput: {
+		borderBottomWidth: 1,
+		borderColor: '$gray300',
+		paddingHorizontal: 20,
+		paddingVertical: 10,
+		width: 250,
+	},
+	passwordBox: {
+		width: Dimensions.get('screen').width - 40,
+		backgroundColor: '$white',
+		borderColor: '$gray300',
+		alignItems: 'center',
+		justifyContent: 'space-between',
+		borderRadius: 10,
+		borderWidth: 1,
+		height: 100,
+		paddingBottom: 15,
+	},
+	restorePassword: {
+		padding: 10,
+		color: '#00f',
+	},
+	saveSettings: {
+		marginVertical: 20,
+	}
+});
+
+export default class Settings extends React.PureComponent<any> {
+	public state = {
+		isAlertVisible: false,
+	};
+
+	private readonly callAlert = () => {
+		Alert.alert(
+			'Confirm new settings',
+			'Are you sure?',
+			[
+				{text: 'Confirm', onPress: () => this.setState({isAlertVisible: false})},
+				{
+					text: 'Cancel',
+					onPress: () => this.setState({isAlertVisible: false}),
+					style: 'cancel',
+				},
+			],
+			{cancelable: false},
+		);
+	}
+
+	private readonly switchRenderer = (type: string, index: number, arr: string[]) => {
+		const style = EStyleSheet.child(styles, 'switch', index, arr.length);
+
+		return (
+			<View
+				key={index}
+				style={style}
+			>
+				<Text>{`${type.charAt(0).toUpperCase()}${type.slice(1)}`}</Text>
+				<Switch
+					value={this.props.temperatureScale === type}
+					onValueChange={() => this.props.changeTemperatureScale(type)}
+				/>
+			</View>
+	)};
+
+	public render() {
+
+		return (
+			<KeyboardAvoidingView
+				style={styles.container}
+				behavior='padding'
+				enabled
+			>
+				{this.state.isAlertVisible ? this.callAlert() : null}
+				<ScrollView>
+					<View style={styles.user}>
+						<Text style={{fontSize: 24}}>Settings</Text>
+					</View>
+					<View style={styles.settingsBox}>
+						<Text style={styles.settingsBoxHeader}>Temperature scale:</Text>
+						{['fahrenheit', 'celsius'].map(this.switchRenderer)}
```

Comment:
{'fahrenheit' && <SomeComponent />}
{'celsius' && <SomeOtherComponent />}

Reply:
why? we must have a all options to switch